go/cmd/dolt/commands/schcmds/tags.go

```diff
@@ -87,24 +89,58 @@ func (cmd TagsCmd) Exec(ctx context.Context, commandStr string, args []string, d
 			return 0
 		}
 	}
-	for i := 0; i < len(tables); i++ {
-		tables[i] = fmt.Sprintf("'%s'", tables[i])
+
+	var headerSchema = sql.Schema{
+		{Name: "table", Type: sql.Text, Default: nil},
+		{Name: "column", Type: sql.Text, Default: nil},
+		{Name: "tag", Type: sql.Uint64, Default: nil},
 	}
 
-	//TODO: implement REGEXP_SUBSTR in go-mysql-server and use it here instead of SUBSTR, as this will eventually break
-	queryStr := fmt.Sprintf("SELECT table_name AS 'table', column_name AS 'column', "+
-		"SUBSTR(extra, 5) AS tag FROM information_schema.columns WHERE table_name IN (%s)", strings.Join(tables, ","))
+	rows := make([]sql.Row, 0)
+
+	for _, tableName := range tables {
+		table, ok, err := root.GetTable(ctx, strings.ToLower(tableName))
+
+		if err != nil {
+			return 1
+		}
+
+		if ok {
+			sch, err := table.GetSchema(ctx)
+
+			if err != nil {
+				return 1
+			}
 
-	if formatStr, ok := apr.GetValue(commands.FormatFlag); ok {
-		return commands.SqlCmd{}.Exec(ctx, "", []string{
-			fmt.Sprintf(`--%s=%s`, commands.FormatFlag, formatStr),
-			fmt.Sprintf(`--%s`, commands.QueryFlag),
-			queryStr + ";",
-		}, dEnv)
+			_ = sch.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+				currRow := sql.NewRow(strings.ToLower(tableName), col.Name, tag)
+
+				rows = append(rows, currRow)
+
+				return false, err
+			})
+		}
+	}
+
+	formatSr, ok := apr.GetValue(commands.FormatFlag)
+
+	var err error
+	if ok {
+		resultFormat, verr := commands.GetFormat(formatSr)
+
+		if verr != nil {
+			return commands.HandleVErrAndExitCode(errhand.VerboseErrorFromError(verr), usage)
+		}
+
+		err = commands.PrettyPrintResults(ctx, resultFormat, headerSchema, sql.RowsToRowIter(rows...))
 	} else {
-		return commands.SqlCmd{}.Exec(ctx, "", []string{
-			fmt.Sprintf(`--%s`, commands.QueryFlag),
-			queryStr + ";",
-		}, dEnv)
+		// Default to tabular.
```

Comment:
Instead of if / else here, you should declare an outputFmt var with default of tabular, then override it if the format flag was passed.

Basically reduce the scope of the conditional logic to just the one var that changes, rather than having a call to PrettyPrint with different params on two different conditional branches.